docs/types/reference-types.rst

```diff
@@ -197,6 +197,8 @@ the ``.push`` member functions are not available).
 You either have to calculate the required size in advance
 or create a new memory array and copy every element.
 
+Elements of newly allocated arrays are always initialized with the :ref:`default value<default-value>`.
+
```

Comment:
I added this only for dynamically allocated arrays because a user was confused specifically whether dynamically allocated memory is zero-initialized. I see more places where this clarification might be useful (basically all reference types) but we don't usually explicitly mention that. Should I add it in those places too?

Reply:
Maybe you could at least change the wording of this sentence. Currently it reads as if this would be a special thing instead of something that is always the case for each and every variable in solidity.